src/app/CommandSender.h

```diff
@@ -36,15 +36,69 @@
 #include <system/SystemPacketBuffer.h>
 
 #include <app/Command.h>
+#include <app/MessageDef/CommandPath.h>
+#include <app/MessageDef/StatusElement.h>
 
 #define COMMON_STATUS_SUCCESS 0
 
 namespace chip {
 namespace app {
 
-class CommandSender : public Command, public Messaging::ExchangeDelegate
+class CommandSender final : public Command, public Messaging::ExchangeDelegate
 {
 public:
+    // TODO: Move various CommandResponse related callbacks to this delegate.
+    class Delegate
```

Comment:
The term 'delegate' is way too over-used in the SDK, and doesn't convey precisely what this interface is.

Suggestion:
```suggestion
    class ResponseHandlerInterface
```